app/application/services/session_auth_service.py

```
@@ -0,0 +1,198 @@
+                ttl=self._lock_ttl,
+                wait_seconds=self._lock_wait_seconds,
+            ) as acquired:
+                if not acquired:
+                    if await self._wait_and_restore(
+                        client=client,
+                        cache_key=cache_key,
+                        credential=credential,
+                    ):
+                        return
+                else:
+                    if await self._try_restore_cache(
```

Ruff (0.15.14)

[warning] Use `elif` instead of `else` then `if`, to reduce indentation

Convert to `elif`

(PLR5501)